haystack/tools/component_tool.py

```diff
@@ -0,0 +1,296 @@
+            field_description = f"Field '{field.name}' of '{cls.__name__}'."
+            if isinstance(schema["properties"], dict):
+                schema["properties"][field.name] = self._create_property_schema(field.type, field_description)
+        return schema
+
+    def _create_pydantic_schema(self, python_type: Any, description: str) -> Dict[str, Any]:
+        """
+        Creates a schema for a Pydantic model.
+
+        :param python_type: The Pydantic model type.
+        :param description: The description of the model.
+        :returns: A dictionary representing the Pydantic model schema.
+        """
+        schema = {"type": "object", "description": description, "properties": {}}
+        required_fields = []
+
+        for m_name, m_field in python_type.model_fields.items():
+            field_description = f"Field '{m_name}' of '{python_type.__name__}'."
+            if isinstance(schema["properties"], dict):
+                schema["properties"][m_name] = self._create_property_schema(m_field.annotation, field_description)
+            if m_field.is_required():
+                required_fields.append(m_name)
+
+        if required_fields:
+            schema["required"] = required_fields
+        return schema
+
+    def _create_basic_type_schema(self, python_type: Any, description: str) -> Dict[str, Any]:
+        """
+        Creates a schema for a basic Python type.
+
+        :param python_type: The Python type.
+        :param description: The description of the type.
+        :returns: A dictionary representing the basic type schema.
+        """
+        type_mapping = {str: "string", int: "integer", float: "number", bool: "boolean", dict: "object"}
+        return {"type": type_mapping.get(python_type, "string"), "description": description}
+
+    def _create_property_schema(self, python_type: Any, description: str, default: Any = None) -> Dict[str, Any]:
+        """
+        Creates a property schema for a given Python type, recursively if necessary.
+
+        :param python_type: The Python type to create a property schema for.
+        :param description: The description of the property.
+        :param default: The default value of the property.
+        :returns: A dictionary representing the property schema.
+        """
+        nullable = self._is_nullable_type(python_type)
+        if nullable:
+            non_none_types = [t for t in get_args(python_type) if t is not type(None)]
+            python_type = non_none_types[0] if non_none_types else str
+
+        origin = get_origin(python_type)
+        if origin is list:
+            schema = self._create_list_schema(get_args(python_type)[0] if get_args(python_type) else Any, description)
+        elif is_dataclass(python_type):
+            schema = self._create_dataclass_schema(python_type, description)
+        elif is_pydantic_v2_model(python_type):
+            schema = self._create_pydantic_schema(python_type, description)
```

Comment:
Our components don't use Pydantic Models, so I’m unsure why we’re supporting this.
Allowing such flexibility might be counterproductive.

If there's an important perspective or requirement that I'm overlooking, I’d appreciate it if you could elaborate.